js/usercss.js

```diff
@@ -0,0 +1,311 @@
+/* global loadScript mozParser semverCompare */
+
+'use strict';
+
+// eslint-disable-next-line no-var
+var usercss = (function () {
+  const METAS = [
+    'author', 'description', 'homepageURL', 'icon', 'license', 'name',
+    'namespace', 'noframes', 'preprocessor', 'supportURL', 'var', 'version'
+  ];
+
+  const BUILDER = {
+    default: {
+      postprocess(sections, vars) {
+        let varDef = ':root {\n';
+        for (const key of Object.keys(vars)) {
+          varDef += `  --${key}: ${vars[key].value};\n`;
+        }
+        varDef += '}\n';
+
+        for (const section of sections) {
+          section.code = varDef + section.code;
+        }
+      }
+    },
+    stylus: {
+      preprocess(source, vars) {
+        return loadScript('vendor/stylus-lang/stylus.min.js').then(() => (
+          new Promise((resolve, reject) => {
+            let varDef = '';
+            for (const key of Object.keys(vars)) {
+              varDef += `${key} = ${vars[key].value};\n`;
+            }
+
+            // eslint-disable-next-line no-undef
+            stylus(varDef + source).render((err, output) => {
+              if (err) {
+                reject(err);
+              } else {
+                resolve(output);
+              }
+            });
+          })
+        ));
+      }
+    }
+  };
+
+  const colorParser = (function () {
+    const el = document.createElement('div');
+    // https://bugs.webkit.org/show_bug.cgi?id=14563
+    document.head.appendChild(el);
+
+    function _parse(color) {
+      const [r, g, b, a = 1] = color.match(/[.\d]+/g).map(Number);
+      return {r, g, b, a};
+    }
+
+    function parse(color) {
+      el.style.color = color;
+      if (el.style.color === '') {
+        throw new Error(chrome.i18n.getMessage('styleMetaErrorColor', color));
+      }
+      color = getComputedStyle(el).color;
+      el.style.color = '';
+      return _parse(color);
+    }
+
+    function format({r, g, b, a = 1}) {
+      return `rgba(${r}, ${g}, ${b}, ${a})`;
+    }
+
+    function pad(s) {
+      if (s.padStart) {
+        // chrome 57+
+        return s.padStart(2, '0');
```

Comment:
srsly man! no need for this hipster stuff, simply rework the code e.g. using the simple single expression I've shown recently with binary shifts. We're using the same code that runs on our minimally supported Chrome version, which is 49.